plexapi/utils.py

```diff
@@ -718,3 +720,14 @@ def sha1hash(guid):
 
 def cleanXMLString(s):
     return _illegal_XML_re.sub('', s)
+
+
+def parseXMLString(s: str):
+    """ Parse an XML string and return an ElementTree object. """
+    if not s.strip():
+        return None
+    encoded_s = s.encode('utf-8')
+    try:  # Attempt to parse the string as-is without cleaning (which is expensive)
+        return ElementTree.fromstring(encoded_s)
+    except ElementTree.ParseError:  # If it fails, clean the string and try again
+        return ElementTree.fromstring(cleanXMLString(encoded_s))
```

Comment:
Does `cleanXMLString` work on the encoded string? The original function cleaned then encoded.